src/extension.ts

```diff
@@ -523,6 +519,14 @@ export function activate(context: vscode.ExtensionContext) {
     context.subscriptions
   );
 
+  vscode.workspace.onDidCloseTextDocument((document: vscode.TextDocument) => {
```

Comment:
This is used to determine if an untitled document was closed and filter out any highlights associated with it.

Reply:
Ok cool I understand this now. Thanks!